desktop/package/package.gradle

```diff
@@ -394,6 +396,57 @@ task packageInstallers {
             executeCmd(jPackageFilePath + commonOpts + linuxOpts +
                     " --linux-rpm-license-type AGPLv3" + // https://fedoraproject.org/wiki/Licensing:Main?rd=Licensing#Good_Licenses
                     " --type rpm")
+
+
+
+            // Define Flatpak-related properties
+            String flatpakManifestFile = 'package/linux/exchange.haveno.Haveno.yml'
+            String linuxDir = 'package/linux'
+            String flatpakOutputDir = 'package/linux/build'
+            String flatpakExportDir = "${binariesFolderPath}/fpexport"
+            String flatpakBundleFile = "${binariesFolderPath}/haveno.flatpak"
+
+            // Read the default app name from the HavenoExecutable.java file
+            def filer = file('../core/src/main/java/haveno/core/app/HavenoExecutable.java')
+            def content = filer.text
+            def matcher = Pattern.compile(/public static final String DEFAULT_APP_NAME = "(.*?)";/).matcher(content)
+            def defaultAppName = "Haveno"
+            if (matcher.find()) {
+                defaultAppName = matcher.group(1)
+            } else {
+                throw new GradleException("DEFAULT_APP_NAME not found in HavenoExecutable.java")
+            }
+            // copy the manifest to a new tmp one in the same place
+            // and add a --filesystem=.local/share/${name} to the flatpak manifest
+            def manifest = file(flatpakManifestFile)
+            def newManifest = file('exchange.haveno.Haveno.yaml')
+            newManifest.write(manifest.text.replace("- --share=network", "- --share=network\n  - --filesystem=~/.local/share/${defaultAppName}:create"))
+            flatpakManifestFile = 'exchange.haveno.Haveno.yaml'
+
```

Comment:
Please remove 3 newlines, so there's only one extra newline, since it's all related to flatpak.